cluster/mesos/cluster.go

```diff
@@ -168,12 +170,20 @@ func NewCluster(scheduler *scheduler.Scheduler, TLSConfig *tls.Config, master st
 
 // Handle callbacks for the events
 func (c *Cluster) Handle(e *cluster.Event) error {
+	// publish event to the watchQueue for external subscribers
+	c.watchQueue.Publish(e)
```

Comment:
@nishanttotla That still doesn't explain why this isn't just a handler. You can do this with a closure to adapt `Publish` to `Handler`.